File: python/sample/sample_builder.py
#! /usr/bin/python
#! -*- coding: utf-8 -*-

import sys
import unittest
import os
import time
from random import randint

sys.path.append(os.path.join(os.path.dirname(__file__), '../src'))
import SimpleTokenBuilder
import AccessToken

appID = "970CA35de60c44645bbae8a215061b33"
appCertificate = "5CFd2fd1755d40ecb72977518be15d3b"
channelName = "7d72365eb983485397e3e3f9d460bdda"
uid = "2882341273"
expiredTs = 1446455471
salt = 1
ts = 1111111

def main():
  
  builder = SimpleTokenBuilder.SimpleTokenBuilder(appID, appCertificate, channelName, uid)
  builder.token.salt = salt
  builder.token.ts = ts
  builder.token.messages[AccessToken.kJoinChannel] = expiredTs

  '''
  print builder.token.messages
  builder.initPriviliges(Role_Admin)
  print builder.token.messages
  builder.setPrivilege(kJoinChannel, expiredTs)
  print builder.token.messages
  builder.removePrivilege(kJoinChannel)
  print builder.token.messages
  '''
  
  result = builder.buildToken()
  print "Access Token:", result

if __name__ == "__main__":
  main()

File: nodejs/sample/sample_builder.js
var SimpleTokenBuilder = require('../src/SimpleTokenBuilder').SimpleTokenBuilder;
var Role = require('../src/SimpleTokenBuilder').Role;
var Priviledges = require('../src/AccessToken').priviledges;
var appID  = "970CA35de60c44645bbae8a215061b33";
var appCertificate     = "5CFd2fd1755d40ecb72977518be15d3b";
var channel = "7d72365eb983485397e3e3f9d460bdda";
var uid = 2882341273;
var salt = 1;
var ts = 1111111;
var expiredTs = 1446455471;

var builder = new SimpleTokenBuilder(appID, appCertificate, channel, uid);
builder.key.salt = salt;
builder.key.ts = ts;
builder.key.messages[Priviledges.kJoinChannel] = expiredTs;

// console.log(builder.key.messages);
// builder.initPriviliges(Role.kRoleAttendee);
// console.log(builder.key.messages);
// builder.setPrivilege(Priviledges.kJoinChannel, expiredTs);
// console.log(builder.key.messages);
// builder.removePrivilege(Priviledges.kJoinChannel);
// console.log(builder.key.messages);

var token = builder.buildToken();
console.log(token);
